warehouse/warehouse.go

```diff
@@ -2028,8 +2028,9 @@ func startWebHandler(ctx context.Context) error {
 	}
 
 	srv := &http.Server{
-		Addr:    fmt.Sprintf(":%d", webPort),
-		Handler: bugsnag.Handler(mux),
+		Addr:              fmt.Sprintf(":%d", webPort),
+		Handler:           bugsnag.Handler(mux),
+		ReadHeaderTimeout: 3 * time.Second,
```

Comment:
I am curious, how did we come up with this number? Should we make this configurable and keep the default values as 3 seconds?